resolve/config.go

```diff
@@ -77,8 +77,9 @@ func (_ *Configurer) KnownDirectives() []string {
 func (_ *Configurer) Configure(c *config.Config, rel string, f *rule.File) {
 	rc := getResolveConfig(c)
 	rcCopy := &resolveConfig{
-		overrides: rc.overrides[:],
+		overrides: make([]overrideSpec, len(rc.overrides)),
```

Comment:
Instead of unconditionally allocating a new slice here, let's do this:

```
rcCopy := &resolveConfig{
  overrides: rc.overrides[:len(rc.overrides):len(rc.overrides)],
}
```

That way, we'll use the original backing array most of the time, but we'll allocate a new array if we append anything. Most build files won't have `# gazelle:resolve` directives, so this should save some memory.